middleware/jwt-middleware.ts

```diff
@@ -0,0 +1,46 @@
+import * as express from "express";
+import JWTAuthentication from "../lib/jwt-authentication";
+
+declare global {
+    namespace Express {
+        export interface Request {
+            // The publicKey that was used to validate the presented JWT
+            // Ensure that authenticated operations are only performed for this publicKey
+            authenticatedPublicKey?: string
```

Comment:
This allows any downstream controller to call `request.authenticatedPublicKey` to get the currently authenticated public key, and ensure that someone can't validate one publicKey, but operate on another.